src/SymfonyContainerBridge.php

```diff
@@ -85,10 +86,12 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
             }
 
             return $entry;
-        } catch (NotFoundException $e) {
+        } catch (NotFoundExceptionInterface $e) {
             if ($invalidBehavior === self::EXCEPTION_ON_INVALID_REFERENCE) {
                 throw new ServiceNotFoundException($id, null, $e);
             }
+        } catch (ContainerExceptionInterface $e) {
+            throw new \Exception('Error while retrieving the entry.', $e->getCode(), $e);
```

Comment:
Why catch a more specific exception to rethrow it with a less precise one? You could just let the original exception bubble (by not catching it)?

Reply:
Because here you don't really respect Symfony's contract (which itself doesn't respect Psr/Container's contract).
Symfony's contract expects `ServiceCircularReferenceException`, `ServiceNotFoundException` and `Exception` to be thrown.

`ContainerExceptionInterface` can be thrown by `$this->fallbackContainer->get($id)`.
But `ContainerExceptionInterface` does not extend or implement anything expected by Symfony.

Reply:
If I understand the problem: `ContainerExceptionInterface` doesn't extend/implement `\Exception`.

That's correct, but it's impossible to throw anything that doesn't extend exception so I think catching + rethrowing isn't necessary 🤔

Reply:
Maybe :)

Reply:
But honestly, I don't if it would work in more "structured" language. For instance, I'm not sure it would compile in Java